doc/adr/0004-structure-convention.md

```diff
@@ -0,0 +1,68 @@
+# 4. Structure convention
+
+Date: 2024-09-24
+
+## Status
+
+Accepted
+
+## Context
+
+The core packages gains more and more logic, we need a way to have small piece of code that are easily maintainable.
+
+## Decision
+
+### Domain Driven
+Code is split by domain. Each domain will have its own directory, and within each directory, the code will be further organized by functionality. This structure will help in maintaining a clean and manageable codebase.
+
+### Naming Convention
+
+To maintain consistency across the codebase, we will adopt a clear and concise naming convention. All files and directories will use `kebab-case` for their names. Except for main or index file, each file will be named using the format `name-of-file.name-of-folder.ts`, ensuring uniformity and readability. 
+
+#### Examples
+
+- Directories: `data-type-templates`
+- Files: `queries.data-type-templates.ts`
+
+### Type Declaration Files
+
+Type declaration files are special TypeScript files that end with the `.d.ts` extension. These files are used exclusively for declaring types and do not contain any executable code. By using `.d.ts` files, we ensure that type definitions are centralized, easily accessible, and separate from implementation code.
```

Comment:
Should the `.d.ts` files be placed within the same domain directories, or should there be a centralized directory for all type declarations?

Reply:
Yep in the same directory as the domain. I'll update the ADR to make it more clear.